### ISSUE TITLE
Track more liveness in "autoscaling stuck"

### ISSUE THREAD

**Issue by sharnoff** [neondatabase/autoscaling#770]
## Problem description / Motivation

Currently, "autoscaling stuck" metrics and logs use the following definition:

> An autoscaling-enabled VM is "stuck" if there has not been a successful health check response for the last 20s.

Current implementation is [here](https://github.com/neondatabase/autoscaling/blob/a7943db25bf5d97eaf4771b46e10a3d97ed4b3ea/pkg/agent/globalstate.go#L511-L528).

This misses various other ways that autoscaling may currently be failing for a particular VM, some of which we've seen in prod (e.g. due to missing a pod start event, the scheduler doesn't know about a particular VM and always returns 404 to the autoscaler-agent).

## Feature idea(s) / DoD

Some other types of "stuckness" we should look at including:

- Requests to the scheduler plugin are failing
- Requests to update the VM object are failing
- _Other_ requests to the vm-monitor are failing

And possibly also (although more difficult):

- Scheduler plugin consistently denying desired upscaling
- vm-monitor consistently denying desired downscaling (currently sometimes expected in practice)

See also #594. Tracking the delay between VM object change and VM status change is currently blocked on #592.

**Comment by shayanh** [comment]
The mentioned types of VM stuckness is covered by other metrics:

* Requests to the scheduler plugin are failing: covered by [Agent → Scheduler plugin request errors](https://www.notion.so/neondatabase/Agent-Scheduler-plugin-request-errors-1a96a43891e244bbb62dfc50a7ef7e7f?pvs=4)
* Requests to update the VM object are failing: covered by [Agent → NeonVM API request errors](https://www.notion.so/neondatabase/Agent-NeonVM-API-request-errors-bd705636bb684d408f96889948a6c23f?pvs=4)
* Other requests to the vm-monitor are failing [Agent → vm-monitor request errors](https://www.notion.so/neondatabase/Agent-vm-monitor-request-errors-179c292d99484b3abe6836e8bab20b62?pvs=4)

I understand this is more fine-grained in a per-VM granularity level, but there is some duplication, particularly, if we are looking forward to setup alerts based on this. @sharnoff wdyt?

**Comment by shayanh** [comment]
@sharnoff and I discussed this. It's a nice to have for observability but not strictly necessary since it's already covered by other metrics. I'm going to move this to optional part of https://github.com/neondatabase/company_projects/issues/187 and keep working on it.

**Comment by Omrigan** [comment]
Status: after the deployment to prod yesterday there is a non-zero number of stuck VM [on the dashboard](https://neonprod.grafana.net/d/ddci7lz4hbo5cd/neon-autoscaling-stuck-vms?orgId=1&from=1713886773043&to=1713973173044). 

For that VMs vm-monitor consistently rejects downscaling request, and those VMs are now considered stuck.
The question is: do we want to investigate why autoscaler-agent wants to downscale VMs, but vm-monitor rejects the request, or we just want to consider this situation normal, and remove it from the definition of stuckness? 
   
Thread: https://neondb.slack.com/archives/C03F5SM1N02/p1713893262823009

**Comment by stradig** [comment]
Implementation finished, some follup in #926
